### PR TITLE
Optimize the isPlenum method

### PR DESCRIPTION
**Pull request by macumber**

### CODE REVIEW

**Comment by macumber** [comment]
@kbenne, @asparke2 noted that this method was a performance bottleneck

File: openstudiocore/src/model/ThermalZone.cpp [hunk 1]
```diff
@@ -1998,18 +1998,13 @@ namespace detail {
   {
     boost::optional<AirLoopHVACSupplyPlenum> result;
 
-    std::vector<AirLoopHVACSupplyPlenum> plenums = model().getModelObjects<AirLoopHVACSupplyPlenum>();
+    std::vector<AirLoopHVACSupplyPlenum> plenums = getObject<ModelObject>().getModelObjectSources<AirLoopHVACSupplyPlenum>();
```
Comment (mbadams5) [inline openstudiocore/src/model/ThermalZone.cpp]: @macumber should use `model().getConcreteModelObjects<AirLoopHVACSupplyPlenum>()` instead, it'll be higher performance and is probably driving most of slowness here.

File: openstudiocore/src/model/ThermalZone.cpp [hunk 2]
```diff
@@ -1998,18 +1998,13 @@ namespace detail {
...
-    {
-      if( boost::optional<ThermalZone> tz = plenum.thermalZone() )
-      {
-        if( tz->handle() == handle() )
-        {
-          result = plenum;
-          break;
-        }
-      }
+    if (plenums.size() == 1){
```
Comment (mbadams5) [inline openstudiocore/src/model/ThermalZone.cpp]: @macumber If you switch to using `getConcreteModelObjects` you'll need revert to using the old approach.

File: openstudiocore/src/model/ThermalZone.cpp (hunk 1 above)
Comment (macumber) [inline openstudiocore/src/model/ThermalZone.cpp]: Either one should be faster, we could experiment or just pick one, I don't have a big preference

Comment (kbenne) [inline openstudiocore/src/model/ThermalZone.cpp]: Never been a fan of seeing the method getModelObjectSources in use. If the performance gain can be achieved using getConcreteModelObjects but leaving the method the same, I'm more in favor of that.

**Comment by nrel-bot-3** [comment]
@kbenne @macumber @lgentile it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@kbenne @macumber @lgentile it has been 7 days since this pull request was last updated.

**Pull request merged**